src/transport/quinn.rs

```diff
@@ -211,23 +218,30 @@ struct ClientChannelInner {
     sender: flume::Sender<oneshot::Sender<Result<SocketInner, quinn::ConnectionError>>>,
 }
 
-impl Drop for ClientChannelInner {
+impl Drop for ClientConnectionInner {
     fn drop(&mut self) {
-        tracing::debug!("Dropping client channel");
+        tracing::debug!("Dropping client connection");
         if let Some(endpoint) = self.endpoint.take() {
-            endpoint.close(0u32.into(), b"client channel dropped");
+            tracing::debug!("Closing endpoint");
```

Comment:
Optional: could I tempt you to use a span for this? Rather than let you drown in the docs here's an example, you need to attach the span to an entire future which is no problem here: https://github.com/n0-computer/iroh/blob/fe97e4d154b0c15007f067241c6d2961ea346f4e/src/provider.rs#L331-L351

Reply:
On the subject of spans I am a bit burned. Really sucks if your instrumentation is the cause for some issues in production...

See

https://github.com/tokio-rs/tracing/issues/515
https://github.com/tokio-rs/tracing/issues/458

But will do. I am sure things got better...